sanic/app.py

```diff
@@ -1032,11 +1033,21 @@ async def handle_request(self, request):
 
     @property
     def test_client(self):
-        return SanicTestClient(self)
+        if self._test_client:
+            return self._test_client
+        from sanic_testing.testing import SanicTestClient  # type: ignore
```

Comment:
```suggestion
        elif hasattr(self, "_test_manager"):
            return self._test_manager.test_client  # type: ignore
        from sanic_testing.testing import SanicTestClient  # type: ignore
```

Reply:
Let's add that in another PR. I want to get this one merged.